compiler/modulegraphs.nim

```diff
@@ -55,10 +55,12 @@ type
     concreteTypes*: seq[FullId]
     inst*: PInstantiation
 
-  SymInfoPair* = object
+  SymInfoPair* = ref object
```

Comment:
Does it really need to be `ref`? The compiler suffers from a massive overuse of `ref` making things harder to understand than necessary.

Reply:
How do we set caughtExceptions then? This is done in a different pass, after the SymInfoPair objects have been collected. Then, we need to update them, by setting caughtExceptions. If it's not a ref object, we should copy the entire object, change a single field, then copy it again? Isn't that worse than using a 'ref object'? Or, is there an easier way?

Reply:
Why not update it in-place or `move` the object?

Reply:
Ok: https://github.com/nim-lang/Nim/pull/23202/commits/9ff6cd988a2e1351e1f4ef5d0fa6ad7ba15ee935